src/main/java/io/github/syst3ms/skriptparser/expressions/ExprColorValues.java

```diff
@@ -0,0 +1,96 @@
+package io.github.syst3ms.skriptparser.expressions;
+
+import io.github.syst3ms.skriptparser.Parser;
+import io.github.syst3ms.skriptparser.lang.Expression;
+import io.github.syst3ms.skriptparser.lang.TriggerContext;
+import io.github.syst3ms.skriptparser.lang.properties.PropertyExpression;
+import io.github.syst3ms.skriptparser.parsing.ParseContext;
+import io.github.syst3ms.skriptparser.util.color.Color;
+import org.jetbrains.annotations.Nullable;
+
+import java.math.BigInteger;
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * Certain color values of a given color.
+ *
+ * @name Date Values
+ * @type EXPRESSION
+ * @pattern [the] (0:rgb (value|color)|1:red [value]|2:green [value]|3:blue [value]) of %color%
+ * @pattern %color%'[s] (0:rgb (value|color)|1:red [value]|2:green [value]|3:blue [value])
+ * @pattern %color% as rgb
+ * @since ALPHA
+ * @author Mwexim
+ */
+public class ExprColorValues extends PropertyExpression<BigInteger, Color> {
+	static {
+		Parser.getMainRegistration().addPropertyExpression(
+				ExprColorValues.class,
+				BigInteger.class,
+				false,
+				"color",
+				"(0:rgb (value|color)|1:red value|2:green value|3:blue value)"
+		);
+		Parser.getMainRegistration().addExpression(
+				ExprColorValues.class,
+				BigInteger.class,
+				false,
+				"%color% as rgb"
```

Comment:
We didn't already added a parse system for types to avoid this syntax? If we didn't, we should implement it and use it.

Reply:
You’re confusing types and this actual syntax.
There’s indeed a syntax called ExprParseAs to parse a String as a type, but this is different. You have to note that a color is already in rgb (red, green blue), but the default display is as a hex. This means that this expression is needed to get the 3 integer values of this color.